Utilities/bootstrap

```diff
@@ -428,13 +438,48 @@ def build_tsc(args):
 
     build_with_cmake(args, cmake_flags, args.tsc_source_dir, args.tsc_build_dir)
 
+def build_yams(args):
+    note("Building Yams")
+    args.yams_build_dir = os.path.join(args.target_dir, "yams")
+
+    cmake_flags = []
+    if platform.system() == 'Darwin':
+        cmake_flags.append("-DCMAKE_C_FLAGS=-target x86_64-apple-macosx%s" % g_macos_deployment_target)
+        cmake_flags.append("-DCMAKE_OSX_DEPLOYMENT_TARGET=%s" % g_macos_deployment_target)
+
+    build_with_cmake(args, cmake_flags, args.yams_source_dir, args.yams_build_dir)
+
+def build_swift_driver(args):
+    note("Building SwiftDriver")
+    args.swift_driver_build_dir = os.path.join(args.target_dir, "swift-driver")
+
+    cmake_flags = [
+        get_llbuild_cmake_arg(args),
+        "-DTSC_DIR=" + os.path.join(args.tsc_build_dir, "cmake/modules"),
+        "-DYams_DIR=" + os.path.join(args.yams_build_dir, "cmake/modules"),
+    ]
+    if platform.system() == 'Darwin':
+        cmake_flags.append("-DCMAKE_C_FLAGS=-target x86_64-apple-macosx%s" % g_macos_deployment_target)
+        cmake_flags.append("-DCMAKE_OSX_DEPLOYMENT_TARGET=%s" % g_macos_deployment_target)
+
+    build_with_cmake(args, cmake_flags, args.swift_driver_source_dir, args.swift_driver_build_dir)
+
+def add_rpath_for_cmake_build(args, rpath):
+    "Adds the given rpath to the CMake-built swift-build"
+    swift_build = os.path.join(args.bootstrap_dir, "bin/swift-build")
+    add_rpath_cmd = ["install_name_tool", "-add_rpath", rpath, swift_build]
```

Comment:
Would it be possible to do this with `-rpath` arguments to the linker instead?  I realize that this change is in inline with the existing approach, but I've been investigating some intermitted failures that can occur when there isn't enough space in the Mach-O header for the additional load commands.  The approach I was going to use was to switch to passing `-rpath` at link time, which seems cleaner.  Alternatively, we would need to make sure that the link command leaves enough space by using `-headerpad`, but that's a bit messier since we'd have to pick a number large enough to make sure there's enough space, and it would leave an unnecessary hole in the binary.

Reply:
It should be possible, yeah; we just need to figure out how to drive CMake to do that. That should be a separate pull request, but I could certainly see the argument that we should implement that improvement before merging here, because adding a bunch more rpaths in this manner is likely to push us over the limit.